rust_src/src/lisp.rs

```diff
@@ -258,6 +258,19 @@ impl From<Vec<LispObject>> for LispObject {
     }
 }
 
+impl<T> From<Vec<T>> for LispObject
+where
+    LispObject: From<T>,
+{
+    default fn from(v: Vec<T>) -> LispObject {
+        list(
```

Comment:
If we're keeping the generic version and the specialized `Vec<LispObject>` version, then should the generic one be `LispObject::from(v.into_iter()...)` (that is, convert the `Vec<T>` into `Vec<LispObject>` and then convert that into `LispObject`)? Otherwise the `list(&...)` logic is being duplicated.

Reply:
I explicitly duplicated it to ensure we did not recurse and cause issues.

Reply:
Which may be an unfounded concern. But I saw recursive explosion in some of my attempts to get this working this morning.

Reply:
My thinking is since we are saying the `Vec<LispObject>` case is a specialization it seems odd to have the default implementation depend on it.

Reply:
If there are better arguments then I can be swayed.

Reply:
My mental model of what's going on here is mostly based on the behavior of Python objects plus whatever I feel like the compiler should be able to figure out.